pkg/helper/secretutils.go

```diff
@@ -197,3 +197,44 @@ func IsSecretWatchedBy3scale(secret *v1.Secret) bool {
 
 	return false
 }
+
+func IsSecretWatchedBy3scaleBySecretName(client k8sclient.Client, secretName, namespace string) bool {
+	secret := &v1.Secret{}
+	secretKey := k8sclient.ObjectKey{
+		Name:      secretName,
+		Namespace: namespace,
+	}
+	err := client.Get(context.TODO(), secretKey, secret)
+	if err != nil {
+		return false
+	}
+
+	existingLabels := secret.Labels
+	if existingLabels != nil {
+		if _, ok := existingLabels["apimanager.apps.3scale.net/watched-by"]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+func ValidateRedisURLPrefix(redisUrl string, isTLS bool) error {
+	if redisUrl == "" {
```

Comment:
we had a discussion about this block @valerymo 
Please refactor the whole function, I think all this should be doing is something like:

if !isTLS {
    if  !strings.HasPrefix(redisUrl, "redis://") {
          then fail, otherwise it's all good.
          }
  }
  
  Reason for this is when you have Sentinels enabled with TLS, your not necessarily need the "rediss" as part of the URL.

Reply:
yes finished function should look like 
```go
  func ValidateRedisURLPrefix(redisUrl string, isTLS bool) error {
	if redisUrl == "" {
		return fmt.Errorf("Redis URL cannot be empty in secret")
	}
	// If TLS is enabled, URL should start with "rediss://"
	if !isTLS {
		// If TLS is not enabled, URL should start with "redis://"
		if !strings.HasPrefix(redisUrl, "redis://") {
			return fmt.Errorf("invalid URL, when TLS is not enabled, URL must start with 'redis://', " +
				" also confirm your port matches the Non-TLS port set in your redis.conf.")
		}
	}
	return nil
}
```

Reply:
@MStokluska , as I see it - for Client - if Sentinel  - Client is talking with Sentinel instead of Server. It's same for Client. So need have it as is. We can discuss it again. cc @austincunningham

Reply:
discussed, updated, thank you @MStokluska , @austincunningham

Reply:
Problem with case when No Sentinel, and when swithing from NonTLS to TLS in APIM, but secret contains NonTLS definition like `redis://172.30.133.42:6379/1`. In this case - r system-app-pre Job issue. Error - unrecoverable. We need find different solution for Sentinel. cc @austincunningham @MStokluska

Reply:
Reverted to the initial solution of the ValidateRedisURLPrefix function with a modification: added isSentinel parameter.
- If TLS is enabled and Sentinel is not used, the Redis URL should start with rediss://.
- If Sentinel is used, the Redis URL becomes irrelevant, as the client communicates with Redis Sentinel rather than the Redis master.
Validation has been done for various cases, including scenarios with Sentinel.